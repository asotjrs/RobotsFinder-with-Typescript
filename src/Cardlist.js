import React from "react";
import Card from "./Card";

const Cardlist=({input})=>{
    const users=input.split(" ");
    const renderedList=users.map((userName)=>{
        return <Card input={userName} key={userName}/>;

    });
    return <div>{renderedList}</div>;



};
export default Cardlist;