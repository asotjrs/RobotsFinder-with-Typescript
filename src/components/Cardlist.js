import React from "react";
import Card from "./Card";

const Cardlist=({robots})=>{
    console.log(robots);
    const users=robots.split(",");
    const renderedList=users.map((userName,id)=>{
        return <Card robot={userName.trim()} key={id}/>;

    });
    return <div>{renderedList}</div>;



};
export default Cardlist;