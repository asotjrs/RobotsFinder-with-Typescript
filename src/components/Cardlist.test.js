
import {shallow} from 'enzyme';
import React from "react";
import Cardlist from "./Cardlist";
import toJson from "enzyme-to-json";
it('to match snap-shot', function () {
    const mockrobots="this,is,just,to,test";
    expect(toJson(shallow(<Cardlist robots={mockrobots}/>))).toMatchSnapshot();

});


