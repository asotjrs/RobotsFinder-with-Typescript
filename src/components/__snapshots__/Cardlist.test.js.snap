// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snap-shot 1`] = `
<div>
  <Card
    key="0"
    robot="this"
  />
  <Card
    key="1"
    robot="is"
  />
  <Card
    key="2"
    robot="just"
  />
  <Card
    key="3"
    robot="to"
  />
  <Card
    key="4"
    robot="test"
  />
</div>
`;
